client/blocks/nav-unification-modal/index.jsx

```diff
@@ -0,0 +1,127 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { Guide } from '@wordpress/components';
+import { Title } from '@automattic/onboarding';
+import { useSelector, useDispatch } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { savePreference } from 'calypso/state/preferences/actions';
+import { getPreference, hasReceivedRemotePreferences } from 'calypso/state/preferences/selectors';
+import { getCurrentUserId } from 'calypso/state/current-user/selectors';
+import { useTranslate } from 'i18n-calypso';
+
+import './style.scss';
+
+const Page = ( { heading, content, image } ) => {
+	return (
+		<div className="nav-unification-modal__page">
+			<div className="nav-unification-modal__text">
+				<div className="nav-unification-modal__heading">
+					<Title tagName="h2">{ heading }</Title>
+				</div>
+				<div className="nav-unification-modal__description">
+					<p>{ content }</p>
+				</div>
+			</div>
+			<div className="nav-unification-modal__visual">{ image }</div>
+		</div>
+	);
+};
+
+const Modal = () => {
+	const dispatch = useDispatch();
+	const userId = useSelector( ( state ) => getCurrentUserId( state ) );
+	const hasPreferences = useSelector( hasReceivedRemotePreferences );
```

Comment:
Would you also need to ensure preferences are queried? 🤔  (There's `QueryPreferences` component.)